herbert_core/classes/MPIFramework/messages/MESS_NAMES.m

```diff
@@ -174,77 +245,99 @@
-                is = false;
-            end
-        end
-        %
         function is = tag_valid(the_tag)
             % verify if the tag provided is valid message tag.
-            if the_tag>=1 && the_tag <= numel(MESS_NAMES.mess_names_)
-                is = true;
-            else
-                is = false;
-            end
+            mn = MESS_NAMES.instance();
+            is = isKey(mn.tag_to_name_map_,the_tag);
         end
         %
-        function is= is_blocking(mess_name)
+        function is= is_blocking(mess_name_or_tag)
             % check if the message with the name, provided as imput is
             % blocking message. (should be send-received synchroneously)
             %
             % Input:
             % mess_name -- a string with message name or cellarray of
-            %              message names
+            %              message names, or a number defining the message
+            %              tag, or cellarray of the messages or array of
+            %              tags or cellarray of message classes.
             % Output
             % is        -- logical array, containing true if the corresponend
             %              message is blocking and false otherwise
             %
-            [~,blocking] = MESS_NAMES.mess_factory();
-            name2code_map = MESS_NAMES.name_tag_maps();
-            if iscell(mess_name)
-                ids = cellfun(@(x)(name2code_map(x)+2),mess_name,'UniformOutput',true);
-                is = blocking(ids);
-            elseif isnumeric(mess_name)
-                is = blocking(mess_name+2);
-            else
-                id = name2code_map(mess_name)+2;
-                is = blocking(id);
+            if isa(mess_name_or_tag,'aMessage')
+                is = mess_name_or_tag.is_blocking;
+                return
             end
+            
+            mni = MESS_NAMES.instance();
+            if isnumeric(mess_name_or_tag)
+                if numel(mess_name_or_tag) > 1
+                    is = arrayfun(@(mn)mni.tag_to_name_map_(mn),mess_name_or_tag,...
+                        'UniformOutput',true);
+                else
+                    name = mni.tag_to_name_map_(mess_name_or_tag);
+                    mc = mni.mess_class_map_(name);
+                    is = mc.is_blocking();
+                end
+            elseif ischar(mess_name_or_tag)
+                mc = mni.mess_class_map_(mess_name_or_tag);
+                is = mc.is_blocking();
+            elseif iscell(mess_name_or_tag)
+                is = cellfun(@(mn)MESS_NAMES.is_blocking(mn),mess_name_or_tag,...
+                    'UniformOutput',true);
+            end
+            
         end
         %
-        function names = get_all_names()
-            % return all message names, subscribed to the factory
-            names  = MESS_NAMES.mess_names_;
+        function is = is_persistent(mess_or_name_or_tag)
+            % check if given message is a persistent message
```

Comment:
it's unclear from this whether it is working on one or more messages and returning one or an array of bools

Reply:
modified